test/unit/horsewhisperer_test.cpp

```diff
@@ -292,13 +292,81 @@ TEST_CASE("parse", "[parse]") {
         REQUIRE(HW::GetFlag<bool>("alias") == true);
     }
 
-    SECTION("it parses and sets aliased action flagsflags") {
+    SECTION("it parses and sets aliased action flags") {
         HW::DefineActionFlag<bool>("test-action", "a alias", "aliased flag", false, nullptr);
         const char* args[] = { "test-app", "test-action", "-a"};
         REQUIRE(HW::Parse(3, const_cast<char**>(args)) == HW::ParseResult::OK);
         REQUIRE(HW::GetFlag<bool>("a") == true);
         REQUIRE(HW::GetFlag<bool>("alias") == true);
     }
+
+    SECTION("it parses and sets integer numbers") {
+        HW::DefineGlobalFlag<int>("int-flag", "no useful description", 42, nullptr);
+
+        SECTION("positive") {
+            SECTION("value after space") {
+                const char* args[] = { "test-app", "test-action", "--int-flag", "3"};
+                REQUIRE(HW::Parse(4, const_cast<char**>(args)) == HW::ParseResult::OK);
+            }
+
+            SECTION("key=value format") {
+                const char* args[] = { "test-app", "test-action", "--int-flag=3"};
+                REQUIRE(HW::Parse(3, const_cast<char**>(args)) == HW::ParseResult::OK);
+            }
+
+            REQUIRE(HW::GetFlag<int>("int-flag") == 3);
+        }
+
+        SECTION("negative") {
+            SECTION("value after space") {
+                const char* args[] = { "test-app", "test-action", "--int-flag", "-4"};
+                REQUIRE(HW::Parse(4, const_cast<char**>(args)) == HW::ParseResult::OK);
+            }
+
+            SECTION("key=value format") {
+                const char* args[] = { "test-app", "test-action", "--int-flag=-4"};
+                REQUIRE(HW::Parse(3, const_cast<char**>(args)) == HW::ParseResult::OK);
+            }
+
+            REQUIRE(HW::GetFlag<int>("int-flag") == -4);
```

Comment:
Would it be better to move this into the sections? We're overwriting the result in HW with the 2nd call and that means if the first test passes but gives us the wrong result, the 2nd section will fix it.